### PR TITLE
Feat/condo/doma 5012/new schema bank integration context

### PR DESCRIPTION
**Pull request by AntonAL**
Stores usage of specific integration by specific organization.

See "banking" section in architecture presentation.

Information about last synchronization time will be obtained from upcoming schema `BankSyncTask`

### CODE REVIEW

File: apps/condo/domains/banking/schema/BankIntegrationContext.test.js [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        adminClient = await makeLoggedInAdminClient()
+        supportClient = await makeClientWithSupportUser()
+        anonymousClient = await makeClient()
+        bankIntegration = await BankIntegration.getOne(adminClient, { id: BANK_INTEGRATION_IDS['1CClientBankExchange'] })
+    })
+
+    describe('CRUD tests', () => {
+        describe('create', () => {
+            test('admin can', async () => {
+                const admin = await makeLoggedInAdminClient()
```
Comment (VKislov) [inline apps/condo/domains/banking/schema/BankIntegrationContext.test.js:35]: `beforeAll` already has `adminClient`, maybe you should use it here

**Review by VKislov — APPROVED** [review]

**Comment by VKislov** [comment]
Comments on not  significant points, so I’ll immediately put an approve

File: apps/condo/domains/banking/schema/BankAccount.js [hunk 2]
```diff
@@ -24,6 +26,33 @@ const BankAccount = new GQLListSchema('BankAccount', {
...
+                validateInput: async ({ existingItem, resolvedData, addFieldValidationError, operation }) => {
+                    const newItem = { ...existingItem, ...resolvedData }
+                    const bankIntegrationContext = await getById('BankIntegrationContext', get(newItem, 'integrationContext'))
+                    const existingBankAccounts = await find('BankAccount', { integrationContext: { id: bankIntegrationContext.id } })
+                    if (existingBankAccounts.length > 1) {
+                        return addFieldValidationError(`Multiple BankAccount with ids "[${map(existingBankAccounts, 'id').join(', ')}]" are connected to BankIntegrationContext(id="${bankIntegrationContext.id}")`)
+                    }
+                    if (operation === 'create') {
+                        if (existingBankAccounts.length > 0) {
+                            return addFieldValidationError(`Cannot connect to BankIntegrationContext, used by another BankAccount(id="${existingBankAccounts[0].id}")`)
```
[the remark this answers is not in the document]
Comment (AntonAL) [reply apps/condo/domains/banking/schema/BankAccount.js:44]: I know, the problem here is that GQLError does not supports array of errors, which is a result of Keystone schemas validation logic.

File: apps/condo/domains/banking/access/BankIntegrationContext.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+const { throwAuthenticationError } = require('@open-condo/keystone/apolloErrorFormatter')
+const { getById } = require('@open-condo/keystone/schema')
+const { checkPermissionInUserOrganizationOrRelatedOrganization } = require('../../organization/utils/accessSchema')
+const { queryOrganizationEmployeeFor, queryOrganizationEmployeeFromRelatedOrganizationFor } = require('@condo/domains/organization/utils/accessSchema')
+
+async function canReadBankIntegrationContexts ({ authentication: { item: user } }) {
+    if (!user) return throwAuthenticationError()
+    if (user.deletedAt) return false
+
+    if (user.isAdmin) return {}
```
[the remark this answers is not in the document]
Comment (AntonAL) [reply apps/condo/domains/banking/access/BankIntegrationContext.js:15]: Semantically, empty object `{}` means that read query gets no restriction changes from access control function.

**Pull request merged**